### PR TITLE
refactor: Crawler/cloud functions

### LINKED ISSUE #208: Subtasks: Implement crawl factory for CloudFunctions resource.
#153

### PR DESCRIPTION
**Pull request by peb-peb**
closes #208 
ref to #153 

@mxmssh @ZetaTwo

### CODE REVIEW

File: src/gcp_scanner/crawler/cloud_functions_crawler.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      while request is not None:
+        response = request.execute()
+        functions_list = response.get("functions", [])
+        request = service.projects().locations().functions().list_next(
+            previous_request=request, previous_response=response)
+    except Exception:
+      logging.info("Failed to retrieve CloudFunctions for project %s", project_id)
+      logging.info(sys.exc_info())
+
+    return functions_list
```
Comment (ZetaTwo) [inline src/gcp_scanner/crawler/cloud_functions_crawler.py:52]: Add newline

Comment (peb-peb) [reply src/gcp_scanner/crawler/cloud_functions_crawler.py]: Done :+1:

**Comment by ZetaTwo** [comment]
Still merge conflicts

**Comment by peb-peb** [comment]
Resolved the merge conflicts!

**Review by ZetaTwo — APPROVED** [review]

**Pull request merged**